scarb/src/ops/package.rs

```diff
@@ -185,6 +231,21 @@ fn prepare_archive_recipe(pkg: &Package) -> Result<ArchiveRecipe> {
         contents: ArchiveFileContents::Generated(Box::new(|| Ok(VERSION.to_string().into_bytes()))),
     });
 
+    // Add VCS info file.
+    if has_vcs(pkg) {
```

Comment:
if you `Clone` `pkg`, you can do this:

```suggestion
    if let Ok(repo) = PackageRepository::open(pkg) {
```

and then you can _move_ repo into the closure

this saves you a `gix::discover` call.